File: src/main/java/com/atsuishio/superbwarfare/entity/mixin/ExplosionAccess.java
package com.atsuishio.superbwarfare.entity.mixin;

import net.minecraft.world.level.Explosion;

public interface ExplosionAccess {

    static ExplosionAccess of(Explosion explosion) {
        return (ExplosionAccess) explosion;
    }

    float superbwarfare$getRadius();
    double superbwarfare$getX();
    double superbwarfare$getY();
    double superbwarfare$getZ();
}

File: src/main/kotlin/com/atsuishio/superbwarfare/entity/mixin/ExplosionAccess.kt
package com.atsuishio.superbwarfare.entity.mixin

import net.minecraft.world.level.Explosion

@Suppress("FunctionName")
interface ExplosionAccess {
    fun `superbwarfare$getRadius`(): Float

    fun `superbwarfare$getX`(): Double

    fun `superbwarfare$getY`(): Double

    fun `superbwarfare$getZ`(): Double

    companion object {
        fun of(explosion: Explosion): ExplosionAccess {
            return explosion as ExplosionAccess
        }
    }
}
